lib/engine/game/g_1870/step/track.rb

```diff
@@ -18,21 +19,15 @@ def lay_tile(action, extra_cost: 0, entity: nil, spender: nil)
           end
 
           def process_lay_tile(action)
-            old_tile = action.hex.tile
-
             super
 
-            old_tile.label = nil if %i[yellow green].include?(old_tile.color)
-            old_tile.label = 'P' if old_tile.id == '170'
-
-            return if action.tile.color == 'gray'
+            return if action.tile.color != :brown
 
-            if action.tile.hex.id == 'B11'
-              action.tile.label = 'K P'
-            elsif action.tile.hex.id == 'C18'
-              action.tile.label = 'P L'
-            elsif %w[J3 J5 N17].include?(action.tile.hex.id)
-              action.tile.label = 'P'
+            case action.tile.hex.id
+            when 'B11'
+              action.tile.future_label = Engine::Part::FutureLabel.new('K', 'gray')
```

Comment:
I merged your changes into the 18Mex future labels PR and rack rake passes with both sets together.

I looked back at 1870 and the only way I can see to make it to avoid this kind of logic in game code would require rendering changes to make it possible to display more than one LargeIcon and show more than one FutureLabel for Kansas City and St. Louis.

In the meantime it might be worth changing the K and L labels to "KC" and "StL" to help link the gray tiles to which map hexes they end up going on before the brown P tile gets laid